### PR TITLE
LAU-948 encryption of reason field on database

### PR DESCRIPTION
**Pull request by priyankaVerma21**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/LAU-948

### Change description ###
encryption of reason field on database

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestFindRepository.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        query.setFirstResult((int) pageable.getOffset());
+        query.setMaxResults(pageable.getPageSize());
+
+        final List<AccessRequest> results = query.getResultList();
+        long totalCount = countResults(encryptionKey, accessRequest);
+
+        return new PageImpl<>(results, pageable, totalCount);
+    }
+
+    private long countResults(final String encryptionKey,
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestFindRepository.java:71]: encryptionKey is unused in countResults

File: src/main/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestInsertRepository.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            reason)
+        VALUES (:requestType, :userId, :caseRef, :action, :logTimestamp, :requestStartTimestamp, :requestEndTimestamp,
+            encode(pgp_sym_encrypt(cast(:reason as text), cast(:encryptionKey as text)), 'base64'))
+            RETURNING id;
+        """;
+
+    @PersistenceContext
+    private final EntityManager entityManager;
+
+    public AccessRequestInsertRepository(final EntityManager entityManager) {
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestInsertRepository.java:25]: You could just use @AllArgsConstructor instead?

File: src/main/java/uk/gov/hmcts/reform/laubackend/cases/service/AccessRequestService.java [hunk 3]
```diff
@@ -24,19 +28,36 @@
 public class AccessRequestService {
 
     private final AccessRequestRepository accessRequestRepository;
+    private final AccessRequestInsertRepository accessRequestInsertRepository;
+    private final AccessRequestFindRepository accessRequestFindRepository;
     private final QueryBuilder queryBuilder;
+    private final TimestampUtil timestampUtil;
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/laubackend/cases/service/AccessRequestService.java:34]: This is unused

**Review by MartinYSpasov — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestFindRepository.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+import uk.gov.hmcts.reform.laubackend.cases.domain.AccessRequest;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+@SuppressWarnings("unchecked")
+@Repository
+public class AccessRequestFindRepository {
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestFindRepository.java:21]: I don't see any GET/POST repository tests

**Review by MartinYSpasov — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/laubackend/cases/service/AccessRequestService.java [hunk 5]
```diff
@@ -24,19 +27,35 @@
...
+
     public AccessRequestLog save(final AccessRequestLog accessRequestLog) {
-        AccessRequest saved = accessRequestRepository.save(accessRequestLog.toModel());
-        return AccessRequestLog.modelToDto(saved);
+        final AccessRequest accessRequest;
+        if (BooleanUtils.isTrue(encryptionEnabled)) {
+            accessRequest = accessRequestInsertRepository
+                .saveAccessRequestAuditWithEncryption(accessRequestLog.toModel(), securityDbBackendEncryptionKey);
+        } else {
+            accessRequest = accessRequestRepository.save(accessRequestLog.toModel());;
```
Comment (rapolaskaseliscgi) [inline src/main/java/uk/gov/hmcts/reform/laubackend/cases/service/AccessRequestService.java:50]: Rogue semi-colon at the end.

File: src/test/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestRepositoryTest.java [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+import static org.springframework.data.domain.PageRequest.of;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@TestPropertySource(properties = {
+    "spring.jpa.hibernate.ddl-auto=update",
+    "spring.liquibase.enabled=false",
+    "spring.flyway.enabled=true"
+})
+public class AccessRequestRepositoryTest {
```
Comment (rapolaskaseliscgi) [inline src/test/java/uk/gov/hmcts/reform/laubackend/cases/repository/AccessRequestRepositoryTest.java:30]: Please, remove this `public` modifier.

**Pull request merged**